pkg/controlplane/controller/leaderelection/leaderelection_controller_test.go

```diff
@@ -747,9 +752,9 @@ func TestController(t *testing.T) {
 								continue // only candidate-aware controllers will follow preferredHolder
 							}
 
-							t.Logf("Deleting lease %s/%s because of preferredHolder %q != %q", l.Namespace, l.Name, *ph, *l.Spec.HolderIdentity)
+							fmt.Printf("Deleting lease %s/%s because of preferredHolder %q != %q", l.Namespace, l.Name, *ph, *l.Spec.HolderIdentity)
 							if err = client.CoordinationV1().Leases(expectedLease.Namespace).Delete(ctx, expectedLease.Name, metav1.DeleteOptions{}); err != nil {
-								t.Logf("Error deleting lease %s/%s: %v", l.Namespace, l.Name, err)
+								fmt.Printf("Error deleting lease %s/%s: %v", l.Namespace, l.Name, err)
```

Comment:
why?

Reply:
We lose correlation of the output with the test this way.
> I got an error that t.Logf may not be called after test ctx is finished. We select on ctx.Done() but during the 10ms or so that the LeaseCandidate update is happening, the test ctx could be cancelled and logging is not permitted. Changed them to printf statements.

Why not wait for them to finish with a wait group?

Reply:
done

Reply:
ah using the wg is a nice trick, thanks!